src/utils/use-interval.ts

```diff
@@ -17,15 +18,32 @@ export function useInterval(
   // Set up the interval.
   useEffect(() => {
     if (disabled) return;
+    const actualDelay = Math.max(delay ?? 0, 100);
+
+    let id: NodeJS.Timeout | number = 0;
+
     function tick() {
-      savedCallback.current();
-    }
-    if (triggerImmediately) {
-      tick();
+      return setTimeout(async function innerCallback() {
+        await savedCallback.current();
+
+        if (!disabled) {
+          tick();
```

Comment:
As an extra safe guard, could it be considered to use `requestAnimationFrame(() => tick());`?

Reply:
I really, really don't think `requestAnimationFrame` should be used as a general timing function outside animations that need to hook into the animation frame. The degree we call useInterval in the codebase, I think, will increase layout paint overhead significantly and I think it will have the opposite effect rather than improving performance.

But this is just my understanding of the effects of these and if everyone else disagrees I won't stand in your way but you'll have to add it as a separate PR to easier revert it after the fact 😂

Reply:
Not going to insist 😛